pkg/discovery/network/generic.go

```diff
@@ -89,7 +97,7 @@ func findClusterIPRangeSvcCreation(clientSet kubernetes.Interface) (string, erro
 
 	// creating invalid svc didn't fail as expected
 	if err == nil {
-		return "", fmt.Errorf("creating invalid service(%v) didn't fail", invalidSvcSpec)
+		return "", nil
```

Comment:
Keeping this return error makes unit tests complex. All existing tests for network plugin detection will need to rely on creating svc always return error.

Reply:
So this means the API server didn't behave as we expected, likely something changed. I think it would be useful to at least log a message:

`status.QueueWarningMessage("Could not determine the service IP range via service creation - the expected error was not returned")`

Reply:
Changed it to return an error properly and changed unit tests as well.